### PR TITLE
feat: onboarding plus step

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Add new step for Plus subscription during onboarding

## Events

Did you introduce any new tracking events?

image

## Experiment

Not anymore, according to [#experiments channel thread](https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1732632687070809)

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://feat-onboarding-plus-step.preview.app.daily.dev

### CODE REVIEW

File: packages/webapp/pages/onboarding.tsx
Comment (sshanzel) [inline packages/webapp/pages/onboarding.tsx:234]: Is this not going to be behind any existing experiment we have for plus subscription?

File: packages/shared/src/components/plus/PlusList.tsx
Comment (sshanzel) [inline packages/shared/src/components/plus/PlusList.tsx:29]:
```suggestion
  'Everything on Free plan',
  'Ads-free browsing',
```

File: packages/shared/src/components/plus/PlusList.tsx
Comment (sshanzel) [inline packages/shared/src/components/plus/PlusList.tsx:40]: I'm not sure about other places, but here, I think the color should be `tertiary`.

Screenshot 2024-11-23 at 2 05 43 PM

File: packages/shared/src/components/plus/PlusComparingCards.tsx
Comment (sshanzel) [inline packages/shared/src/components/plus/PlusComparingCards.tsx:29]: Just for accuracy 😁 
```suggestion
const PlanCard: FC<PlusCardProps> = ({
```

Comment (sshanzel) [reply packages/shared/src/components/plus/PlusComparingCards.tsx:29]: Should we instead have a map of the content should be? For example:
```
const planMap = {
  free: {
    title: 'Free',
    description: 'yada yada',
  },
  plus: {
    title: 'Plus',
   description: 'Some description',
  }
}
```

This is to avoid more ternary operators once we get more plans which is likely to happen.

File: packages/shared/src/components/onboarding/OnboardingPlusStep.tsx
Comment (sshanzel) [inline packages/shared/src/components/onboarding/OnboardingPlusStep.tsx:21]: Looks to be something we can reuse in the future. Might be good to take it to its own file.

**Review by sshanzel — CHANGES_REQUESTED** [review]
Amazing work with the Plus subscription. Things are looking well so far, just had a few comments on some things but nothing major.

Also, kindly wait for review from anyone at Asgard's team as they've already experienced working with the feature 🚀

File: packages/shared/src/components/onboarding/OnboardingHeader.tsx
Comment (rebelchris) [inline packages/shared/src/components/onboarding/OnboardingHeader.tsx:73]: Isn't this the last step actually?
Wondering if it shouldn't just say "create my feed" then, it's what we do for all other steps.

File: packages/shared/src/components/onboarding/OnboardingPlusStep.tsx
Comment (rebelchris) [reply packages/shared/src/components/onboarding/OnboardingPlusStep.tsx:21]: Why FC though 😢

File: packages/shared/src/components/onboarding/OnboardingPlusStep.tsx
Comment (rebelchris) [inline packages/shared/src/components/onboarding/OnboardingPlusStep.tsx:62]: You shouldn't need normal here right if bold is not passed?

File: packages/shared/src/components/plus/PlusComparingCards.tsx
Comment (rebelchris) [inline packages/shared/src/components/plus/PlusComparingCards.tsx:52]:
```suggestion
            className="grid aspect-square size-8 place-content-center rounded-8 bg-surface-float"
```

File: packages/shared/src/components/plus/PlusComparingCards.tsx
Comment (rebelchris) [inline packages/shared/src/components/plus/PlusComparingCards.tsx:87]: Please add `rel={anchorDefaultRel}`

File: packages/webapp/pages/onboarding.tsx
Comment (rebelchris) [reply packages/webapp/pages/onboarding.tsx:234]: sources is a running experiment, you can't rely on it yet.

Comment (rebelchris) [reply packages/webapp/pages/onboarding.tsx:234]: It should actually be it's own evaluation experiment

File: packages/shared/src/components/plus/PlusComparingCards.tsx
Comment (rebelchris) [reply packages/shared/src/components/plus/PlusComparingCards.tsx:87]: I also think this should include the chosen option already right?

**Review by rebelchris — COMMENTED** [review]
Some minor comments

**Comment by ilasw** [comment]
> sources is a running experiment, you can't rely on it yet.

@rebelchris I had to check if the experiment is running because we need to handle the case of users coming from this screen, right? There is some other way to do it?

PS: I think that we could the way to move from one step to another using an array that has some conditional steps, and move within steps using indexes (not a work for this PR still). Wdyt?

```ts

const onboardingSteps = [
  OnboardingStep.Intro, 
  OnboardingStep.Theme, 
  OnboardingStep.EditTag, 
  ...(showOnboardingSources ? [OnboardingStep.Sources] : [])
];

```

File: packages/shared/src/components/onboarding/OnboardingPlusStep.tsx
Comment (ilasw) [reply packages/shared/src/components/onboarding/OnboardingPlusStep.tsx:62]: Bold coming from button outside, so i need to override it for not making it inherited

File: packages/shared/src/components/plus/PlusComparingCards.tsx
Comment (ilasw) [reply packages/shared/src/components/plus/PlusComparingCards.tsx]: Updated ✅

File: packages/shared/src/components/plus/PlusComparingCards.tsx
Comment (ilasw) [reply packages/shared/src/components/plus/PlusComparingCards.tsx:97]: Updated ✅

File: packages/shared/src/components/onboarding/OnboardingHeader.tsx
Comment (rebelchris) [reply packages/shared/src/components/onboarding/OnboardingHeader.tsx:73]: This question is still valid I think?

File: packages/webapp/pages/onboarding.tsx
Comment (rebelchris) [inline packages/webapp/pages/onboarding.tsx:239]: You can actually just add Sources here right?
If either of those 3 in this point of the checks it should add it?

We use early returns so that should be safe I think?

Comment (ilasw) [reply packages/webapp/pages/onboarding.tsx]: Yes, added here! 💯

File: packages/shared/src/hooks/usePlusSubscription.ts
Comment (rebelchris) [inline packages/shared/src/hooks/usePlusSubscription.ts:24]: Oh the only issue here is that it will enroll everyone as soon as they hit onboarding page, and actually anytime this hook gets called even.

Did you check enrollment with ido in #experiment channel?

Comment (ilasw) [reply packages/shared/src/hooks/usePlusSubscription.ts]: Updated ✔️

File: packages/webapp/pages/onboarding.tsx
Comment (rebelchris) [inline packages/webapp/pages/onboarding.tsx:264]: Both these should be enums in log file

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice, just one minor change on enums for logging

File: packages/shared/src/components/plus/PlusComparingCards.tsx
Comment (sshanzel) [reply packages/shared/src/components/plus/PlusComparingCards.tsx]: cc @ilasw

File: packages/shared/src/lib/log.ts
Comment (rebelchris) [inline packages/shared/src/lib/log.ts:287]: I think we have this already, but maybe not id?

**Review by rebelchris — APPROVED** [review]
Approving already, but we should have onboarding already its not a new thing

File: packages/shared/src/lib/log.ts
Comment (ilasw) [reply packages/shared/src/lib/log.ts:287]: We have it on `Origin`, should I use this instead?

**Review by sshanzel — APPROVED** [review]
🚀

**Pull request merged**
